### PR TITLE
ortest: added LUA_52COMPAT and ENABLE_GC64 flags to LuaJIT our builds.

### PR DESCRIPTION
**Pull request by thibaultcha**

### CODE REVIEW

File: samples/ortest-ec2.ob.tt [hunk 1]
```diff
@@ -177,14 +177,14 @@ openssl {
 luajit_sysmalloc {
     always;
     cwd ~/git/luajit2;
-    sh 'sudo rm -rf /opt/luajit-sysm; git reset --hard; git checkout $luajit_branch && git merge origin/$luajit_branch && make CC=gcc47 CCDEBUG=-g XCFLAGS="-DLUAJIT_NUMMODE=2 -DLUAJIT_USE_VALGRIND -I/opt/valgrind/include -DLUAJIT_USE_SYSMALLOC -DLUA_USE_APICHECK -DLUA_USE_ASSERT -msse4.2" -B -j$jobs PREFIX=/opt/luajit-sysm && sudo make install PREFIX=/opt/luajit-sysm';
+    sh 'sudo rm -rf /opt/luajit-sysm; git reset --hard; git checkout $luajit_branch && git merge origin/$luajit_branch && make CC=gcc47 CCDEBUG=-g XCFLAGS="-DLUAJIT_NUMMODE=2 -DLUAJIT_USE_VALGRIND -I/opt/valgrind/include -DLUAJIT_USE_SYSMALLOC -DLUA_USE_APICHECK -DLUA_USE_ASSERT -DLUAJIT_ENABLE_GC64 -DLUAJIT_ENABLE_LUA52COMPAT -msse4.2" -B -j$jobs PREFIX=/opt/luajit-sysm && sudo make install PREFIX=/opt/luajit-sysm';
```
Comment (agentzh) [inline samples/ortest-ec2.ob.tt]: I suggest we  introduce a variable to hold common luajit extra C compiler flags to avoid code duplication :)

**Review by agentzh — CHANGES_REQUESTED** [review]

File: samples/ortest-ec2.ob.tt (hunk 1 above)
Comment (thibaultcha) [reply samples/ortest-ec2.ob.tt]: Done!

File: samples/ortest-ec2.ob [hunk 2]
```diff
@@ -22,6 +22,7 @@ use_lua_resty_core = 0;
 use_stap = 0;
 nginx_cc_opt = '-O1 -I/opt/ssl/include';
 nginx_ld_opt = '-L/opt/ssl/lib';
+luajit_xcflags = '-DLUAJIT_ENABLE_GC64 -DLUAJIT_ENABLE_LUA52COMPAT -msse4.2';
```
Comment (agentzh) [inline samples/ortest-ec2.ob]: `-DLUAJIT_NUMMODE=2 -DLUA_USE_APICHECK -DLUA_USE_ASSERT` should be moved here as well.

**Review by agentzh — CHANGES_REQUESTED** [review]

File: samples/ortest-ec2.ob (hunk 2 above)
Comment (thibaultcha) [reply samples/ortest-ec2.ob]: We don't use those in the openresty-debug target. I will add them, but just making sure this is what we want :)

Comment (thibaultcha) [reply samples/ortest-ec2.ob]: Updated!

Comment (agentzh) [reply samples/ortest-ec2.ob]: @thibaultcha Oh, yeah, openresty's `--with-debug` automatically enables those already, but it doesn't hurt to duplicate those flags anyway :)

**Review by agentzh — APPROVED** [review]

**Pull request merged**